### DISCUSSION TITLE
slinger with python on mac

### DISCUSSION THREAD

**Discussion by f520k510** [GerryDazoo/Slinger#46] [Q&A] [answered]
I tried to set up slinger on mac running python3.  It runs with register successful.
When I tried to view with vlc, the terminal had a message "Register Successful
09/10/2022, 09:42:27   Streaming connection from ('192.168.86.1', 58013)
Hacker Alert. Bad Data from  ('192.168.86.1', 58013)
09/10/2022, 09:42:27   Streaming connection from ('192.168.86.1', 58014)
Hacker Alert. Bad Data from  ('192.168.86.1', 58014)
"
I think it must be a security issue but I don't know how to handle it.

I set up another slinger python on windows 10, with the identical config.ini, and all is fine.
I just had to add exception in windows defender for python.exe.

Please help.  If I can run the server on mac, which is my every day laptop, it would be great without having to run a parallel desktop and a windows 10 on it.

**Comment by GerryDazoo** [comment]
Please supply all the output from program. 

What version are you running. This could be a bug or a mis-configuration. Looks like the code is taking an exception when trying to process the request from VLC. When you say it works on Win10 are you using the same VLC client to connect to MAC and the PC?

Do you feel comfortable editing the python code. We can uncomment a line of debug code to get to the bottom of this quickly.

 

From: f520k510 ***@***.***> 
Sent: Friday, September 09, 2022 9:50 PM
To: GerryDazoo/Slinger ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [GerryDazoo/Slinger] slinger with python on mac (Discussion #46)

 

I tried to set up slinger on mac running python3. It runs with register successful.
When I tried to view with vlc, the terminal had a message "Register Successful
09/10/2022, 09:42:27 Streaming connection from ('192.168.86.1', 58013)
Hacker Alert. Bad Data from ('192.168.86.1', 58013)
09/10/2022, 09:42:27 Streaming connection from ('192.168.86.1', 58014)
Hacker Alert. Bad Data from ('192.168.86.1', 58014)
"
I think it must be a security issue but I don't know how to handle it.

I set up another slinger python on windows 10, with the identical config.ini, and all is fine.
I just had to add exception in windows defender for python.exe.

Please help. If I can run the server on mac, which is my every day laptop, it would be great without having to run a parallel desktop and a windows 10 on it.

**Comment by f520k510** [comment]
I don't know how to enable debug log so I post the output from the terminal:

python3 slingbox_server.py
Version : 3.07 Running on macOS-12.5.1-x86_64-i386-64bit pid= 1775
Using config file config.ini
Connection Manager Running on port 8080 with 10 max streams....
Streamer Running:  10 config.ini 8080
starting up on  port 8080
Checking for slingbox at ('125.228.xx.xx', 5216)
('125.228.xx.xx', 5216) OK
Using slingbox at  ('125.228.xx.xx', 5216)
Streamer:   Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Reading Custom Remote definition from remote.txt
 * Serving Flask app 'slingbox_server'
 * Debug mode: off
WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:8081
 * Running on http://192.168.86.42:8081
Press CTRL+C to quit
09/10/2022, 18:58:19  Registering Slingboxes {'xxxxxxxxxxxEF7CD334': 8080}
Getting external IP address
Redirector, my external IP address 219.79.xx.xx
Registering xxxxxxxxxxxxxxEF7CD334:8080

Register Successful
09/10/2022, 18:59:53   Streaming connection from ('192.168.86.1', 50259)
Hacker Alert. Bad Data from  ('192.168.86.1', 50259)
09/10/2022, 18:59:53   Streaming connection from ('192.168.86.1', 50260)
Hacker Alert. Bad Data from  ('192.168.86.1', 50260)


By uncommenting the python code, if you mean add "//" before a line in the .py file, then I think I can manage to do it.
Yes, I did connect to both mac and win10 with the same vlc player on the mac.

Thanks.

**Reply by GerryDazoo** [reply]
The output from the terminal i what I needed. There is no separate log file.
With a simple text editor like notepad ( don't know what the equivalent is on a MAC) Change line 882
print( 'Hacker Alert. Bad Data from ', client_address )
to
print( 'Hacker Alert. Bad Data from ', client_address, e, traceback.print_exc())

and give it a go. After reviewing the code this looks like something MAC specific on it's TCP implementation. But if we can figure out why it's crashing it should be easy to fix.

**Reply by f520k510** [reply]
Here is the output with the line updated:
I assume you don't need the lines above "register successful".  They are almost identical to the above except the timestamp.  But Please let me know if you need them posted here.

Anyway, here it goes:

Register Successful
09/10/2022, 21:39:53   Streaming connection from ('192.168.86.1', 53421)
Traceback (most recent call last):
  File "/Users/chunchiehliu/Documents/Slinger/slingbox_server.py", line 865, in ConnectionManager
    connection.setsockopt(socket.SOL_SOCKET, socket.SO_SNDBUF, 1024*1024*8)
OSError: [Errno 55] No buffer space available
Hacker Alert. Bad Data from  ('192.168.86.1', 53421) [Errno 55] No buffer space available None
09/10/2022, 21:39:53   Streaming connection from ('192.168.86.1', 53422)
Traceback (most recent call last):
  File "/Users/chunchiehliu/Documents/Slinger/slingbox_server.py", line 865, in ConnectionManager
    connection.setsockopt(socket.SOL_SOCKET, socket.SO_SNDBUF, 1024*1024*8)
OSError: [Errno 55] No buffer space available
Hacker Alert. Bad Data from  ('192.168.86.1', 53422) [Errno 55] No buffer space available None


Thanks for your reply in advance.

**Reply by GerryDazoo** [reply]
great I think this will get you going

line 865
                connection.setsockopt(socket.SOL_SOCKET, socket.SO_SNDBUF, 1024*1024*8)
change to
 #               connection.setsockopt(socket.SOL_SOCKET, socket.SO_SNDBUF, 1024*1024*8)

just comment it out with a # sign. If your just streaming locally this should be OK but if you streaming from a remote media player (i.e. VLC on your cell phone ) you might have some buffering issues. I'll do some googling to see if we can bump up the TCP buffer sizes like I've documented for Linux/RaspberryPi

**Reply by GerryDazoo** [reply]
for some reason Mr. Git made my line bold instead of showing the pound sign at the beginning of the line

**Reply by f520k510** [reply]
Thanks for the quick reply.
I commented out that line, and here is the terminal output:

Register Successful
09/10/2022, 22:22:08   Streaming connection from ('192.168.86.42', 54098)

Slinginfo  "350/500/M1" GdJDpF8SDxPLolP 12 30 125.228.xx.xx 5216 3072
 Starting Stream for  192.168.86.42:54098
Traceback (most recent call last):
  File "/Users/chunchiehliu/Documents/Slinger/slingbox_server.py", line 678, in streamer
    s_ctl, stream  = start_slingbox_session(streams)
  File "/Users/chunchiehliu/Documents/Slinger/slingbox_server.py", line 397, in start_slingbox_session
    s_ctl = sling_open(sling_net_address, 'Control') # open control connection to SB
  File "/Users/chunchiehliu/Documents/Slinger/slingbox_server.py", line 361, in sling_open
    s.setsockopt(socket.SOL_SOCKET, socket.SO_RCVBUF, 1024*1024*8)
OSError: [Errno 55] No buffer space available
 Badness starting slingbox session  [Errno 55] No buffer space available None
zsh: killed     python3 slingbox_server.py

**Reply by GerryDazoo** [reply]
looks like MAC is similar to linux. Try these commands
sysctl -w kern.ipc.maxsockbuf=8192000
sysctl -w net.inet.tcp.sendspace=8192000

I suspect you'll need admin/root privileges to do this.

**Reply by GerryDazoo** [reply]
yes, you'll need to comment out line 361 too.

**Reply by GerryDazoo** [reply]
along with sendspace bump up revcspace
sysctl -w net.inet.tcp.recvspace=8192000

If you can set those kernel parameters you probable don't need to make the server source code changes. Let me know how it works out.  I may need to add a configuration option to allow the user to adjust this.

**Comment by f520k510** [comment, marked as answer]
Thanks for reply again.
I used the commands you gave me :
sysctl -w kern.ipc.maxsockbuf=8192000
This one ran ok.
sysctl -w net.inet.tcp.sendspace=8192000
sysctl -w net.inet.tcp.recvspace=8192000
These two won't run as it says the result is too large.
I trial and error and decided that it would let me enter 7250000 for these two.
Then I ran the slingbox_server.py and it said the same no buffer error.
I looked up your code, and I decided to try changing the values on line 361 and 865 to 7000000.
Because 7,250,000>7,000,000 I guess.

At this point I googled and found these commands, which I decided to give a try:

 //OSX default of 3 is not big enough 
  sysctl net.inet.tcp.win_scale_factor=8
  // increase OSX TCP autotuning maximums
  sysctl net.inet.tcp.autorcvbufmax=33554432
  sysctl net.inet.tcp.autosndbufmax=33554432
  
  I ran slingbox_server.py and this time I could view with vlc just fine.
  Although sometimes the code would break with pipe broken error.
  It ran 2 out of 3 tries so far.
  Sorry I modified your code.
  
  So, a few questions from my side:
  1  Do I have to run sysctl commands every time I restart my mac?
  2  Do the changes I made, i.e. 7,000,000 and 7,250,000 make any sense?
  
  I am running macos monterey 12.5.1, btw.

**Reply by GerryDazoo** [reply]
Thanks for digging into this. You can check if the sysctl -w commands are persistent by looking at the /etc/sysctl.conf file to see if those parameters are there. If not you can add them to the file.
Yes  the changes make sense. Not being a MAC guy I don't have a system to test with but I'm guessing sendspace + revcspace needs to be <= maxsockbuf  based on some googling I did.

Can you check in MAC-land and see if the Linux kernel parameters mentioned in the readme are available? If they are, that might be the way to go.

You should be able to stream all day long with no errors. For a test can you reduce your video Resolution and Videobandwidth to see if it improves things. You may need to do some tweeking. It may be that win8 box can't keep up.

**Reply by GerryDazoo** [reply]
You may also want to try bumping up the priority of the server with the nice command
nice -n -20 python3   slingbox_server.py .....

**Reply by theweatherguy** [reply]
I have to report that modifications to the  /etc/sysctl.conf  are not loaded with Mac Monterey at startup.  The /etc/syctl.conf is **ignored** with recent Mac OS systems.   The settings of sysctl.conf are not only not persistent, but are not loaded.

You can check this by running the terminal command   sysctl -a      It will reveal current settings.   You'll see that your settings in the sysctl.conf file are ignored. 

Indeed, the values you've experimented with get a "too large" error on my Mac when running the command manually.  I am experimenting with different parameter values as a manual command prior starting up the slinger python script.   Relying on the syctl.conf file won't work.  

While slinger works well streaming to my ipad and Mac on my home wireless network, I only get about 5-10 seconds before a disconnect occurs on my iPhone on the cellular network.   I'm using a VPN, not port-forwarding, to connect to my home network when away. 

I suspect the solution is some sort of change in video  parameters for cellular streaming. I know that the regular slingbox app seems to go through a series of incremental tests and optimization in bandwidth when connecting on a cellular network.  The video is often low resolution when you first connect until it optimizes.   

Have you found any video settings that work for cellular network use?

**Reply by f520k510** [reply]
Thanks for pointing it out that these commands are not run after reboot.  I confirm that.  I am running beta version of slinger, which seems to run ok without these commands.  Does it work for you without these modifications and with only commenting out lines 361 and 865?

**Reply by GerryDazoo** [reply]
I'm surprised about the sysctl.conf not being processed. 
Yes, I have two configs, one for streaming HD locally and one for streaming while using Cellular data.
What your phone supports and what your data plan supplies will drive your settings.
I'm using 640x480 resolution and bandwidth set to 1000. YMMV.

**Reply by GerryDazoo** [reply]
PS. A little googling.... Apple moved the defailt localtion for the config file
https://community.jamf.com/t5/jamf-pro/is-etc-sysctl-conf-deprecated-in-macos-catalina/m-p/204843
https://discussions.apple.com/thread/253830342

**Reply by theweatherguy** [reply]
Yes, everything works well streaming locally on home network with only commenting out #361 and 865 as you indicated.

I've been researching and experimenting with setting the sysctl settings as part of the terminal command to launch sling_server.py

I'm not ready to celebrate, but it looks like this appears to be working for cellular, using the VLC app for iPhone:


To launch slinger using Mac Terminal, proceed with _only this_ change—  
 (don't change the receivespace or maxbuffer)

==========
sudo sysctl -w net.inet.tcp.sendspace=5501072
_enter your Mac Admin password_

python3 slingbox_server.py config1.ini config2.ini

===========

(I'm using two different slingboxes; that's the reason for the two config files.)

I using a Keyboard Maestro macro to launch terminal, enter the sysctl command, enter my Admin password  and execute the python3 command 10 seconds after Keyboard Maestro launches.  

If it continues to work well with more testing, I'll update my Mac instructions post.   

Oh, one more thing.  The sysctl change seems to have solved the problem with it working on VLC on my Mac.

**Reply by theweatherguy** [reply]
The command sudo sysctl -w net.inet.tcp.sendspace=5501072  has solved all issues connecting with cellular, wifi or another Mac. It is not persistent and must be re-entered each time you start your Mac or the slingbox_service.py.   And VLC works without any issues now.  Better picture than with the real slingbox app!  I've updated my Mac Step -By-Step  and also made some much needed corrections to typos.

**Comment by f520k510** [comment]
I checked readme.txt and it entails 2 ideas.  One is to be able to satrt or stop slingbox server as a service, the other increase tcp buffer.  I will check both on mac forums to see if slingbox server can be installed as a service and if tcp buffer can be increased that way.

I will try different combinations of video resolution and video bandwidth as well and let you know.

**Comment by JLBla** [comment]
To f520k510, what program did you use on your Mac to use python?

**Reply by f520k510** [reply]
I just install python on python.org.
Then open terminal and run python3 sling box_server.py
After setting up everything like config.ini, port forward, etc, of course.

**Reply by JLBla** [reply]
I am trying to see if there’s any way for me to set up connection using my ipad Pro since this is the device I tend to have with me, and it is a bit hard for me to leave my laptop on at home and connected to this sling server all the time.  I am using Pyto on my ipad to run slingbox_server.py after configuring the config.ini file, but the following shows up:
Version : 3.07 Running on macOS-15.6.1-iPad8,5-64bit pid= 10082
Using config file config.ini
Exception in thread Thread-12 (ConnectionManager):
Traceback (most recent call last):
  File "Pyto.app/site-packages/python3.10/threading.py", line 1009, in _bootstrap_i
nner
    self.run()
  File "Pyto.app/scripts_runner.py", line 64, in run
    super().run()
  File "Pyto.app/site-packages/python3.10/threading.py", line 946, in run
    self._target(*self._args, **self._kwargs)
  File "/private/var/mobile/Containers/Shared/AppGroup/B7CD91A9-ED55-46EA-8C03-2CAD
36176D88/File Provider Storage/Slingbox python/Slinger-main/slingbox_server.py", li
ne 823, in ConnectionManager
    serverinfo = cp['SERVER']
  File "Pyto.app/site-packages/python3.10/configparser.py", line 964, in __getitem_
_
    raise KeyError(key)
KeyError: 'SERVER'

Are these errors something I can fix, or am I doing something wrong using the python file?  I realize this wasn’t probably meant for an ipad, but I thought I’d give it a try.  When I use this config.ini file and the .exe file on my Windows laptop, everything connects fine.  Thanks for your help.

**Reply by GerryDazoo** [reply]
You can't be using the same config.ini file
your missing the [SERVER] section and I suspect everything else....
Did you install all the other required python modules per the readme ( note the latest version does not need the requests module)

**Reply by JLBla** [reply]
I took the zip file and made a new file with all of the files, including the config.ini file and everything else.  I can remove some of the files if that’s the issue?  In the config.ini file, it shows
[SERVER]
; local port number for the server to listen on for connections
port=8080
maxstreams=10
enableremote=no

Is this the [SERVER] section you reference?  Sorry, I am not very familiar with all this information.  I am using the same port 8080 that I was using when I set up my Windows laptop.  Should I be using a different port?

**Reply by GerryDazoo** [reply]
You can't be using that file or you wouldn't be getting the error. The error says you don't have the [SERVER] line in the file passed to the server code. It needs to be a plain text file.

**Reply by JLBla** [reply]
I see what you mean.  I have the files in a directory folder in Pyto, but I guess the config file is not being read with how I have things set up.  I’m not sure how to use this program I guess.

**Reply by GerryDazoo** [reply]
I see no reason why this should not just work. What I'd like you to do is reunzip the zip file and NOT make any changes to the default config.ini and see if we can get over this hump. You won't be able to stream but at least we can make some progress.

**Reply by JLBla** [reply]
Thank you for your help.  So I deleted all the files and reunzipped the zip file into my Downloads folder in my iCloud drive.  I used Pyto and imported the contents of the folder into the directory.  The first time I ran the python file.  I got this message about Flask, so then I changed the config.ini file to read enableremote=no, but still got the same message that I originally wrote about.  I’m not sure what I’m doing wrong.
![F132BA69-6D4E-4B99-B1D7-486516035CE7](https://user-images.githubusercontent.com/113312822/190007239-47e83bcd-ba88-4301-8aa1-10b3ce0ae068.png)

![D297CEB2-5056-4E28-A2E7-D2058EEF7046](https://user-images.githubusercontent.com/113312822/190006891-6021bec7-c598-441a-aa39-646ac8d0139a.png)

**Reply by GerryDazoo** [reply]
The file that I need a dump of is config.ini. It looks like the code can't read the config.ini file. Who knows how iCloud/ipad manages these files? Maybe it's a file permissions issue. Can you confirm the files are stored and running in a directory on the ipad and not from iCloud.
IS there a way for you to run a shell command  "ls -al" in the folder where the files are to confirm?

**Reply by JLBla** [reply]
Ok, I moved the folder from the iCloud to my iPad to see if that made a difference, now it’s a similar message but a different Thread number.  I’m sorry, but I’m not sure how to run a shell command??
![BACC458E-2744-4FF2-AEA9-CAC0DFEDA0F0](https://user-images.githubusercontent.com/113312822/190022305-a429a856-097b-4d66-8d39-bb645e7480df.png)

**Comment by GerryDazoo** [comment]
The thread number will change every time you run the program. It's not significant. I'm still keen on seing the contents of the config.ini on your ipad.

**Reply by JLBla** [reply]
I took images of the config.ini file in Pyto.
![4142E9B8-3E8D-4AA8-9592-8149AC296A08](https://user-images.githubusercontent.com/113312822/190027049-2a6805c9-714f-4af5-a4a6-addec74b482c.png)
![CF3BEA06-F422-44FC-AC86-A9170E63E1BC](https://user-images.githubusercontent.com/113312822/190027053-d3890c28-0943-494d-913b-e63b07c55500.png)

**Reply by JLBla** [reply]
So I was able to get it to run with the config.ini file using Pyto, not sure if it was the right way, but it did connect.  I am now at the part that shows Hacker Alert when I use my ipad to connect using VLC app.  The ipad is at 192.168.2.135
Version : 3.07 Running on macOS-15.6.1-iPad8,5-64bit pid= 15487
Using config file config.ini
Connection Manager Running on port 8080 with 10 max streams....
Streamer Running: starting up on  port 8080 
10 config.ini 8080
Checking for slingbox at ('192.168.2.63', 5301)
('192.168.2.63', 5301) OK
/private/var/mobile/Containers/Shared/AppGroup/B7CD91A9-ED55-46EA-8C03-2CAD36176D88
/File Provider Storage/Slinger-main/Slinger/slingbox_server.py:524: ResourceWarning
: unclosed <socket.socket fd=30, family=AddressFamily.AF_INET, type=SocketKind.SOCK
_STREAM, proto=0, laddr=('192.168.2.135', 51765), raddr=('192.168.2.63', 5301)>
  if not check_ip(sling_net_address): slingip = ''
Using slingbox at  ('192.168.2.63', 5301)
Streamer:  Bedroom Slingbox Waiting for first stream, flushing any IR requests that
 arrive while not connected to slingbox
09/14/2022, 12:53:53   Streaming connection from ('192.168.2.135', 51851)
Hacker Alert. Bad Data from  ('192.168.2.135', 51851)
09/14/2022, 12:53:54   Streaming connection from ('192.168.2.135', 51852)
Hacker Alert. Bad Data from  ('192.168.2.135', 51852)
09/14/2022, 12:53:54   Streaming connection from ('192.168.2.135', 51853)
Hacker Alert. Bad Data from  ('192.168.2.135', 51853)
09/14/2022, 12:53:54   Streaming connection from ('192.168.2.135', 51854)
Hacker Alert. Bad Data from  ('192.168.2.135', 51854)
09/14/2022, 12:53:54   Streaming connection from ('192.168.2.135', 51855)
Hacker Alert. Bad Data from  ('192.168.2.135', 51855)
09/14/2022, 12:53:54   Streaming connection from ('192.168.2.135', 51856)
Hacker Alert. Bad Data from  ('192.168.2.135', 51856)

**Reply by GerryDazoo** [reply]
Look at the original posts in this thread. I suspect you have not properly configured the TCP buffer space. In the meantime comment out lines  361 and 865. Put a # sign at the start of each line.

**Reply by JLBla** [reply]
I am a complete newbie when it comes to these things.  I apologize.  I uncommitted lines 361 and 865 by putting a # sign at the start of each line.  I read about the posts in this thread about trying to add the following lines, but I don’t understand how to do this.  Do I add it to the end of the slingbox_server.py file, or this a separate file I need to make? 
sysctl -w kern.ipc.maxsockbuf=8192000
sysctl -w net.inet.tcp.sendspace=7250000
sysctl -w net.inet.tcp.recvspace=7250000

 I also read some other comment about 
//OSX default of 3 is not big enough
sysctl net.inet.tcp.win_scale_factor=8
// increase OSX TCP autotuning maximums
sysctl net.inet.tcp.autorcvbufmax=33554432
sysctl net.inet.tcp.autosndbufmax=33554432
Do I need to try to do this as well?  Thanks again for your patience and help.

**Reply by GerryDazoo** [reply]
Please reread my last post. Comment out lines 361 and 865 in sling_server.py. Put a # sign at the start of each line. This will get you going until you can figure out how to make the needed OS kernel configuration changes.

**Reply by GerryDazoo** [reply]
BTW, please share how you resolved the issue with not being able to read the config.ini file.

**Reply by JLBla** [reply]
Oops, I just realized I mistyped and wrote that I uncommitted the lines.  I did comment out lines 361 and 865 as you suggested.  I don't get the Hacker alert anymore.  I will wait to see what you are able to come up with for the kernel configuration changes.  

I was able to use Pyto on my iPad with the config.ini file and the py file by opening a Terminal and typing touch slingbox_server.py  
Then I ran the slingbox_server.py afterwards.  I'm not sure why that works, but it does consistently connect if I type that before executing the py file.

**Comment by JLBla** [comment]
To f520k510, how did you do the sysctl commands on the Mac?  I guess Mac’s may have a command prompt or some kind of way??  Any idea how to do it on an iPad Pro?

**Reply by f520k510** [reply]
Just open a terminal and type sysctl xxx.
It is s supported command on macos.
I don't know how to do it on ipad pro.

**Comment by f520k510** [comment]
I think I would sum up what I did to make it works on macos.
add these three lines to /etc/sysctl.conf

sysctl -w kern.ipc.maxsockbuf=8192000
sysctl -w net.inet.tcp.sendspace=7250000
sysctl -w net.inet.tcp.recvspace=7250000
comment out line 361 and 865 of slingbox_server.py by putting a # beore them

**Reply by JLBla** [reply]
Thanks for the info.  I’m pretty sure there isn’t a way to add these modifications on an ipad, since I don’t really have a Terminal that has root access.  I am able to connect to the Sling server with my iPad by commenting out those lines and keeping the media player app on a split screen that can be set larger, rather than half screen.  If I try to stream full screen, it freezes and disconnects frequently.  I have verified also that I can connect remotely with a VPN, and I am able to use python on my iPad with a free app called a-shell.  Thanks for all the help.

**Comment by theweatherguy** [comment]
I just discovered this topic and I’m thrilled that people have gotten GerryDazoo’s incredible Sling hack working on the Mac.  I’m wondering if JLBLa would be willing to post a Mac ‘step by step’ for us total novices of how to get it working on the Mac.   Thank you.


A few specific questions: 

Are you running it on a dedicated Mac?

What Mac system version are you using? Are you running on Monterey?

Do you have to create a separate user and a separate home directory or can it be running from a folder on your regular user account home folder?

**Reply by JLBla** [reply]
I'm sorry.  I'm just reading this post.  I am actually using it on my iPad Pro, so it's a bit different than on a Mac.  I believe f520k510 was able to set it up on a Mac successfully, so perhaps, he can answer your questions.

**Comment by f520k510** [comment]
Yes, I got it working on macos.  I needed to comment out 2 lines in the current release of slinger.
I believe the next release should work withot any modification to it.
However, this may be necessary to get it working:


at the terminal type: sudo nano /etc/sysctl.conf and add these three lines, and then reboot
sysctl -w kern.ipc.maxsockbuf=8192000
sysctl -w net.inet.tcp.sendspace=7250000
sysctl -w net.inet.tcp.recvspace=7250000


When the next release is available, do these: 
You'll need the IP address and port number to use to connect to your remote slingbox. startup a viewing session with the sling software and grab that info from the Network Settings. Unless you really know for sure what there are. On the router where the slingbox resides add a port forwarding rule to map the slingbox port to the internal IP address and port of the slingbox In your config.ini file set the slingbox ipaddress and port number to the external ip address where the slingbox is located. Start the server and start watching.....

**Comment by theweatherguy** [comment]
Hi f520k510,
With your earlier help and additional guidance from Gerry, I believe I have the slingbox_server.py  working.   Thank you so much!

I haven't been able to watch the stream, though.  Which app are you using on the Mac?   Which app are you using on an iPad?

From my internal address, my slingbox_server.py is running at IP address 192.168.2.165:8080/DenSlingbox

Are you using VLC for the Mac or something else?    

Thanks again!

**Reply by theweatherguy** [reply]
Found the answer and got it streaming on my iPad with VNC.

**Reply by JLBla** [reply]
On the iPad, I use VLC.  I’ve tried looking into some other iPad apps, but I can’t find ways to start a network stream in a lot of the other ones.

**Reply by theweatherguy** [reply]
I have an app for iPhone and iPad that works well. It’s called **AV Player**.  There’s  also AV Player HD for iPad. I’ve used this app for years. There's a specific choice to enter a streaming URL.  I’m not seeing any disconnect issues with this app.

**Reply by flipper2222222** [reply]
can you pls tell us, does  AV Player support PIP, Picture in Picture on an Iphone?

**Reply by theweatherguy** [reply]
No, AV Player doesn't support PIP on iPhone or iPad.  

Another discovery and FYI.  I have two iPads.  On my first generation iPad Pro from about 5 or 6 years ago, the stream crashes the app.   On my new M1 iPad Air, it works well but I needed to change the decoder setting in the app settings from "auto" to "S/W".   

On my iPhone11, it works on the home network also with the app decoder setting set to S/W. 

I haven't done port forwarding yet on my router to try this on cellular.   I was hoping to use my WireGuard VPN server setup that I use for secure connections for file/screen sharing on my home network. but I haven't been successful with tests today with slinger on the VPN outside of the home.   Not sure why it didn't connect via the VPN.   Still a work in progress.


****UPDATE***  After field testing, I can't recommend AV Player for this.  I'm having a lack of success maintaining a connection via my cellular network with my iPhone using my  WireGuard VPN server.   It connects with perfect picture for about 8-10 seconds, then disconnects.  I've tried VLC, AV Player, PlayerExtreme.     All connect with perfect picture, then disconnect.   Not sure what the solution is.  Somehow I think there's a lack of adequeate stream buffering or inadequate connection speed. ***

**Reply by GerryDazoo** [reply]
I run my sessions over a VPN during parts of my testing. (Which I can't do for the next few days because my house in Nova Scotia got winged by hurricane Fiona). I think your issue may be who's connecting to who. If your slingbox server is running on the client side of the VPN connection, the server side might not know how to route the packets back to the whole client side network. I had to add a static route on my server side to  be able to route packets back to the client-side using the client VPN address as the gateway.

**Reply by theweatherguy** [reply]
Thank you, Gerry.  That makes sense.  I'll see what I can do to reproduce this.  With WireGuard, I have no problem with file sharing and screen sharing.   

For others thinking about using a VPN, WireGuard is a new VPN server-client for Mac iOS and iPad, Windows, Linux, etc.   Extremely well-reviewed by Steve Gibson of Security Now podcast.  To set up on a home Mac as a Server, I followed the instructions here:    https://barrowclift.me/post/wireguard-server-on-macos   
The iPad-OS Mac and iOS clients are free on the Apple App store.  

Here are some other links that I found useful, bu the first one was a charm:

https://blog.scottlowe.org/2021/04/01/using-wireguard-on-macos/

https://blog.scottlowe.org/2021/06/28/using-wireguard-on-mac-via-cli/

https://blog.scottlowe.org/2021/06/22/making-wireguard-from-homebrew-work-on-an-m1-mac/

**Comment by David1946** [comment]
Have you tried MPV? That is what I now use rather than VLC.

**Comment by GerryDazoo** [comment]
Yes, my streamer of chose is ffplayer due to it being easy to setup a low latency connection.
